File: veriblock-pow-miner/src/main/java/nodecore/miners/pow/ApplicationMeta.java
// VeriBlock PoW CPU Miner
// Copyright 2017-2020 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.
package nodecore.miners.pow;

public class ApplicationMeta {
    public static final String PROGRAM_NAME = ApplicationMeta.class.getPackage().getSpecificationTitle();
    public static final String PROGRAM_VERSION = ApplicationMeta.class.getPackage().getImplementationVersion();
    public static final String FULL_PROGRAM_NAME_VERSION = PROGRAM_NAME + " v" + PROGRAM_VERSION;
}

File: pop-miners/altchain-pop-miner/src/main/kotlin/org/veriblock/miners/pop/ApplicationMeta.kt
// Copyright 2017-2020 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.
package org.veriblock.miners.pop

object ApplicationMeta {
    private val APPLICATION_NAME: String? = ApplicationMeta::class.java.getPackage().specificationTitle
    private val APPLICATION_VERSION: String? = ApplicationMeta::class.java.getPackage().specificationVersion
    val FULL_APPLICATION_NAME_VERSION = "$APPLICATION_NAME v$APPLICATION_VERSION"
}
